app/models/miq_server/worker_management/dequeue.rb

```diff
@@ -61,7 +61,7 @@ def prefetch_stale_threshold
 
   def prefetch_below_threshold?(queue_name, wcount)
     @queue_messages_lock.synchronize(:SH) do
-      return false if @queue_messages[queue_name].nil?
+      return false if @queue_messages[queue_name].nil? || @queue_messages[queue_name][:messages].nil?
```

Comment:
In this case prefer fetch_path

```ruby
return false if @queue_messages.fetch_path(queue_name, :messages).nil?
```

Alternately, if you are saying the key may or may not exist (as opposed to key existing with a nil value), then you can use key_path?

```ruby
return false unless @queue_messages.key_path?(queue_name, :messages)
```